tf/upgrade.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import yaml
+import sys
+import tensorflow as tf
+from tfprocess import TFProcess
+
+START_FROM = 0
+
+def main(cmd):
+    cfg = yaml.safe_load(cmd.cfg.read())
+    print(yaml.dump(cfg, default_flow_style=False))
+
+    root_dir = os.path.join(cfg['training']['path'], cfg['name'])
+    if not os.path.exists(root_dir):
+        os.makedirs(root_dir)
+
+    x = [
+        tf.placeholder(tf.float32, [None, 112, 8*8]),
+        tf.placeholder(tf.float32, [None, 1858]),
+        tf.placeholder(tf.float32, [None, 1])
+    ]
+
+    tfprocess = TFProcess(cfg)
+    tfprocess.init_net(x)
+
+    if os.path.exists(os.path.join(root_dir, 'checkpoint')):
```

Comment:
Is there a use case for running this without starting from a checkpoint? Might be better to fail if no checkpoint found, otherwise it's not going to print anything about creating new names, implying that it upgraded the (non-existing) checkpoint correctly.

Reply:
good idea - done.